src/memray/_memray/logging.h

```diff
@@ -21,6 +21,12 @@ logToStderr(const std::string& message, int level);
 void
 setLogThreshold(int threshold);
 
+logLevel
+getLogThreshold();
+
+#define LOG_IF_DEBUG(msg)                                                                               \
+    if (getLogThreshold() == DEBUG) LOG(DEBUG) << msg
```

Comment:
This should be `>=` technically, though it probably doesn't matter much.

Why not just make `Log::operator<<` handle this? Couldn't we just make it a no-op if the log level is less than the log threshold?

Reply:
Because that still would create the string in the caller, no?

Reply:
No, the signature is:
```cpp
    template<typename T>
    LOG& operator<<(const T& msg)
```
and we're calling it like:
```cpp
LOG_IF_DEBUG(symname << " intercepted!");
```

So `operator<<` is being called twice, each time with a `const char*`. The only place where a dynamic memory allocation is occurring is when `operator<<` does `buffer << msg` because that's appending to a stringstream, which may need to grow the string. If we made `operator<<` return instead of calling `buffer << msg`, no allocation would happen, and we wouldn't need `LOG_IF_DEBUG` instead of our normal `LOG_DEBUG`.

Reply:
Ok, I will implement this in the `operator<<`